sdk/Metrics/ValueRecorder.php

```diff
@@ -0,0 +1,161 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\Sdk\Metrics;
+
+use InvalidArgumentException;
+use OpenTelemetry\Metrics as API;
+
+/*
+ * Name	: ValueRecorder
+ * Instrument kind : Synchronous
+ * Function(argument) : record(value)
+ * Default aggregation : MinMaxSumCount
+ * Notes : Per-request, any non-additive measurement
+ *
+ * ValueRecorder is a non-additive synchronous instrument useful for
+ * recording any non-additive number, positive or negative. Values
+ * captured by a Record(value) are treated as individual events
+ * belonging to a distribution that is being summarized. ValueRecorder
+ * should be chosen either when capturing measurements that do not
+ * contribute meaningfully to a sum, or when capturing numbers that
+ * are additive in nature, but where the distribution of individual
+ * increments is considered interesting.
+ *
+ * Example: One of the most common uses for ValueRecorder is to capture latency
+ * measurements. Latency measurements are not additive in the sense that
+ * there is little need to know the latency-sum of all processed requests.
+ * We use a ValueRecorder instrument to capture latency measurements
+ * typically because we are interested in knowing mean, median, and other
+ * summary statistics about individual events.
+ *
+ * The default aggregation for ValueRecorder computes the minimum and maximum
+ * values, the sum of event values, and the count of events, allowing the rate,
+ * the mean, and range of input values to be monitored.
+ */
+class ValueRecorder extends AbstractMetric implements API\ValueRecorder, API\LabelableMetric
+{
+    use HasLabels;
+
+    /**
+     * @var float $valueSum
+     * @var float $valueMin
+     * @var float $valueMax
+     * @var int $valueCount
+     */
+    protected $valueSum = 0;
+    protected $valueMin = INF;
+    protected $valueMax = -INF;
+    protected $valueCount = 0;
+
+    /*
+     * Testing floating point values for equality is problematic, due to
+     * the way that they are represented internally.  Therefore, force all
+     * precision to a predetermined number, and equality can be determined.
+     */
+    private $decimalPointPrecision = 10;
+
+    /**
+     * getType: get the type of metric instrument
+     *
+     * @return  int
+     */
+    public function getType(): int
+    {
+        return API\MetricKind::VALUE_RECORDER;
+    }
+
+    /**
+     * Returns the sum of the values
+     *
+     * @access	public
+     * @return	float
+     */
+    public function getValueSum(): float
```

Comment:
Should we add all these public functions to the interface?

Reply:
Thank you for reviewing.

I initially left it out as I felt the `API` should mirror the spec and it wasn't covered explicitly in the spec, but looking again, I see:
```
 * Function(argument) : record(value)
 * Default aggregation : MinMaxSumCount
```
Since it calls out the aggregations, I've included functions in the API as there must be a way to access those values.
I didn't include getMean, as that is a nice to have helper function, but isn't strictly needed and isn't called out in the spec as an aggregate either.